### PR TITLE
Download phase

### PR DESCRIPTION
**Pull request by nirs**
Show download phases during download-disk.

Rebased on #110

### CODE REVIEW

**Comment by rokkbert** [comment]
If we use interactive password entry, the progressbar should not be drawn first, otherwise both go in the same line and it looks like this:
image

**Review by rokkbert — CHANGES_REQUESTED** [review]
Looks cool and informative, except for the interactive-password-case.

File: ovirt_imageio/client/_download.py [hunk 1]
```diff
@@ -59,16 +59,17 @@ def register(parser):
 
 
 def download_disk(args):
-    con = _ovirt.connect(args)
-    with closing(con):
-        disk = _ovirt.find_disk(con, args.disk_id)
-        storage_domain = _ovirt.find_storage_domain(con, disk)
-        host = _ovirt.find_host(con, storage_domain.name)
+    with _ui.ProgressBar(phase="creating transfer") as pb:
+        con = _ovirt.connect(args)
```
[the remark this answers is not in the document]
Comment (nirs) [reply ovirt_imageio/client/_download.py:63]: Good catch!

[the remark this answers is not in the document]
Comment (nirs) [reply ovirt_imageio/client/_download.py:63]: Fixed in current version by moving password reading to the parser. This way the input does not clash with the output, and we can create the progress very early so it cover the entire operation.

**Review by aesteve-rh — APPROVED** [review]
Nice!

It would be great to find a way to separate the transfer part from the other phases, so that the speed rate does not decrease in the progress bar updates while finalizing. 

Not for this PR, though!

**Comment by nirs** [comment]
> It would be great to find a way to separate the transfer part from the other phases, so that the speed rate does not decrease in the progress bar updates while finalizing.

True, the rate is relevant relevant for the download phase, but from
user point of view, the total time to download matters. This just show
the horrible inefficiency we have in engine. But with real disks which
may be 100's of GiB and slower storage and network download will take
minutes and few seconds to finalize the transfers are negligible.

If we want to avoid this we can use the progress style of the sdk examples:

    [   0.00 ] creating transfer
    [   0.72 ] downloading image
    [   100% ] 6.00 GiB, 2.67 s, 2.25 GiB/s
    [   2.70 ] finalizing transfer
    [   7.81 ] download completed

With this we get accurate transfer stats, and the total time to download, but the transfer
stats are misleading.

**Pull request merged**